src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -163,6 +173,60 @@ public static SearchContext buildContextForRequest(
       contextBuilder.setHighlightFetchTask(highlightFetchTask);
     }
     contextBuilder.setExtraContext(new ConcurrentHashMap<>());
+
+    if (searchRequest.getInnerHitsCount() > 0) {
+      List<InnerHitFetchTask> innerHitFetchTasks =
+          new ArrayList<>(searchRequest.getInnerHitsCount());
+      for (String innerHitName : searchRequest.getInnerHitsMap().keySet()) {
+        InnerHit innerHit = searchRequest.getInnerHitsOrThrow(innerHitName);
+        Query childQuery =
+            extractQuery(indexState, "", innerHit.getInnerQuery(), innerHit.getQueryNestedPath());
```

Comment:
Since this is an arbitrary query, you will want to use `searcher.rewrite()` on it

Reply:
[indexSearch.search()](https://github.com/apache/lucene/blob/45110a6a46ab8ab6aad271e1c5b55cb6296fdab9/lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java#L514)
will enforce a rewrite and ParentChildrenBlockJoinQuery will rewrite the childQuery

Reply:
That might be ok if the rewrite is not too expensive, since it will happen for each hit

Reply:
get it now. Each hit will rewrite itself if we don't rewrite at the top level.